src/ice_spi.c

```diff
@@ -55,42 +60,29 @@ void ice_spi_init(void) {
     gpio_set_dir(ICE_SPI_SCK_PIN, GPIO_IN);
     gpio_set_dir(ICE_SPI_TX_PIN, GPIO_IN);
     gpio_set_dir(ICE_SPI_RX_PIN, GPIO_IN);
+
+    // Initialize SPI, but don't yet assign the pins SPI function so they stay in high impedance mode.
+    spi_init(spi1, 50 * 1000 * 1000);
```

Comment:
For now, it is fine to hardcode it to something that works for what SPI is most frequently used for.